app/modules/site_settings/models.py

```diff
@@ -466,10 +447,59 @@ def _get_default_value(cls, key):
         return def_val
 
     @classmethod
-    def set_key_value(cls, key, value):
+    def set_after_validation(cls, key, value):
         assert key in cls.HOUSTON_SETTINGS.keys()
         key_data = cls.HOUSTON_SETTINGS[key]
 
+        file_upload_guid = None
+        string = None
+        data = None
+        boolean = None
+        val_type = key_data['type']
+        is_public = key_data.get('public', True)
+        if val_type == dict or val_type == list:
+            data = value
+        elif val_type == bool:
+            assert isinstance(value, bool)
+            boolean = value
+        elif val_type == str:
+            assert isinstance(value, str)
+            string = value
+        elif val_type == 'file':
+            file_upload_guid = value
+        else:
+            # Catch if anyone adds a new 'type' to the houston settings
+            assert False, f'Site setting type {val_type} not supported'
+        log.debug(f'About to update Houston Setting key={key}')
+        existing = cls.query.get(key)
+        if existing:
+            log.debug(f'currently this is {existing}')
+        kwargs = {
+            'key': key,
+            'file_upload_guid': file_upload_guid,
+            'string': string,
+            'data': data,
+            'boolean': boolean,
+            'public': is_public,
+        }
+        setting = cls(**kwargs)
+        with db.session.begin(subtransactions=True):
+            if not cls.query.get(key):
```

Comment:
Note I am uncomfortable with this being an if else but that was what testing showed was required... I think